### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by henrycode460**
This branch contains the followings:

- A ``Nameable``, ``BaseDecorator``, ``CapitalizeDecorator``, and ``TrimmerDecorator`` class
- The ``Person`` inherit the ``Nameable`` class
- The ``CapitalizeDecorator``, and ``TrimmerDecorator`` inherit ``BaseDecorator`` class
- Value from the ``class_name`` method in the  ``CapitalizeDecorator`` output in ``uppercase``
- Value from the ``class_name`` method in the  ``TrimmerDecorator`` output is trimmed to a maximum length of  ``10``

### CODE REVIEW

File: baseDecorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    puts @nameable
+  end
```
Comment (Ismailco) [inline baseDecorator.rb]: - [ ] This method should return @nameable.correct_name, not put the nameable, kindly fix this so it returns the correct value

File: person.rb [hunk 2]
```diff
@@ -13,6 +16,10 @@ def can_use_services?
     of_age? || @parent_permission ? true : false
   end
 
+  def correct_name
+    puts @name
+  end
```
Comment (Ismailco) [inline person.rb]: - [ ] The method should return the @name, not print it to the console, kindly make the changes so the method return the @name

File: capitalizeDecorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative 'baseDecorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    puts @nameable.upcase
+  end
```
Comment (Ismailco) [inline capitalizeDecorator.rb]: - [ ] This method should capitalize only the first letter of the given name not the whole name, kindly consider fixing this so it returns the correct value

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @henrycode460, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative 'nameable'
```
Comment (Nemwel-Boniface) [inline person.rb:1]: - [ ] Trying out your project locally, I got the error saying that the ```correct_name``` method was undefined. On further inspection, I realized that you had not included the following:

image


```
require_relative './baseDecorator'
require_relative './capitalizeDecorator'
require_relative './trimmerDecorator'
```

Despite making this change, I was still not able to run the console and test it correctly. Kindly have a look into why this is the case

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @henrycode460            :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] Kindly have a look into why the console throws an error when I run the tests. The test data I was using I placed it inside the Person.rb file is the following:

```
person = Person.new(22, 'maximilianus')
puts person.correct_name
capitalizedPerson = CapitalizeDecorator.new(person)
puts capitalizedPerson.correct_name
capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
puts capitalizedTrimmedPerson.correct_name
```

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hello @henrycode460 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**